src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/EventLogDataCollector.cs

```diff
@@ -267,6 +266,7 @@ internal string WriteEventLogs(List<EventLogEntry> eventLogEntries, int maxLogEn
         // Write the event log file
         FileTransferInformation fileTransferInformation =
             new(dataCollectionContext, eventLogPath, true, _fileHelper);
+        TPDebug.Assert(_dataSink != null, "Initialize should have been called.");
```

Comment:
We cannot use the initialize pattern we were talking about because this method is called from within the `Initialize` method.